### PR TITLE
[Parallel] Fix missing process RemovedAndAddedFilesProcessor->run() on parallel process on WorkerRunner

### LINKED ISSUE #7231: RemovedAndAddedFilesCollector no longer seems to add files
# Bug Report

<!-- First, thank you for reporting a bug. That takes time and we appreciate that! -->

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | ≥0.13.1             |


With Rector 0.13.0 running a custom Rector rule form https://github.com/j0k3r/graby/pull/293 is able to add files using `RemovedAndAddedFilesCollector`. It stops creating the files after updating Rector to 0.13.1 or newer version, even though the rule still performs the code transformation.  

I do not see anything suspicious in the changes: https://github.com/rectorphp/rector-src/compare/0.13.0...0.13.1.

## Minimal PHP Code Causing Issue

https://github.com/jtojnar/repro/tree/73e3e1813f7ef5ff39484fbe50de13c3c2d13e4b/rector-7231

```patch
1 file with changes
===================

1) test.php:0

    ---------- begin diff ----------
@@ @@
 <?php

+use Test\Foo;
 function test() {
 	new Foo;
 }
    ----------- end diff -----------

Applied rules:
 * TestRector


                                                                                                                        
 [OK] 1 file has been changed by Rector                                                                                 
```

<details>
<summary>It works with Rector 0.13.0</summary>



https://github.com/jtojnar/repro/tree/43c2b8890704e3f27090cb9376ef6b391dbe3ed3/rector-7231

```patch
 ! [NOTE] File "/home/jtojnar/Projects/repro/rector-7231/test" was added                                                

1 file with changes
===================

1) test.php:0

    ---------- begin diff ----------
@@ @@
 <?php

+use Test\Foo;
 function test() {
 	new Foo;
 }
    ----------- end diff -----------

Applied rules:
 * TestRector


 ! [NOTE] 1 files were added                                                                                            

                                                                                                                        
 [OK] 2 files have been changed by Rector                                                                               
```


</details>

## Expected Behaviour

New files should be created.

### PR DESCRIPTION
**Pull request by samsonasik**
@jtojnar this is  on parallel process to:

- Close https://github.com/rectorphp/rector-src/pull/3058
- Fixes https://github.com/rectorphp/rector/issues/7231

### CODE REVIEW

**Comment by samsonasik** [comment]
it works 🎉

**Comment by samsonasik** [comment]
It may need an integration to run in `ApplicationFileProcessor` vs duplicate usage on `PhpFileProcessor`

**Comment by samsonasik** [comment]
When it apply on `PhpFileProcessor`, when parallel disabled back, now got 2 output on disable paralell:

```diff
+! [NOTE] File "./removed-and-added-files-collector/test" will be added
+
+ ! [NOTE] File "./removed-and-added-files-collector/test" will be added
+
+ ! [NOTE] 1 files were added
+
+
```

so definatelly the integration usage is needed.

**Comment by samsonasik** [comment]
Ok, so the parallel work on `WorkerRunner`, so it needs to run under `WorkerRunner` https://github.com/rectorphp/rector-src/pull/3218/commits/2431a596545ba1cdcf26e196c5b1265719dcc3b2

File: e2e/removed-and-added-files-collector/expected-output.diff [hunk 1]
```diff
@@ -0,0 +1,7 @@
+! [NOTE] File
+ !        "./removed-and-added-files-c
+ !        ollector/test" will be added
+
+
+
+ [OK] Rector is done!
```
Comment (samsonasik) [inline e2e/removed-and-added-files-collector/expected-output.diff:7]: this `[OK] Rector is done!` printed message should actually not exists with run with parallel along with:

```php
$this->removedAndAddedFilesProcessor->run($configuration)
```
as it is running with `--dry-run`, but probably need in separate PRs for improvement 👍 :)

**Comment by samsonasik** [comment]
Finally 🎉 All checks have passed 🎉 🎉 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
While I am at it, I applied use `FileProcessorInterface` so all implementation of `FileProcessorInterface` will be applied in parallel proces 🎉 https://github.com/rectorphp/rector-src/pull/3218/commits/b0d97319d6ff4383e7139e487f2bf311758f5c73

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you :partying_face:

**Pull request merged**